tests/irdl/test_attr_constraint.py

```diff
@@ -237,6 +237,28 @@ def test_memref_to_tensor(
         (AnyAttr() & BaseAttr(AttrA), BaseAttr(AttrA)),
         # Note the [Attribute] to provide a supertype of AttrA and Attribute
         (BaseAttr[Attribute](AttrA) & AnyAttr(), BaseAttr(AttrA)),
+        (BaseAttr(AttrA) | BaseAttr(AttrA), BaseAttr(AttrA)),
```

Comment:
This is an instance of the more general: `A | A = A` for any constraint `A` right?

Reply:
I don't see why not, seems like a simpler rule than the one for `BaseAttr` anyway